src/emulator/storage/apis/firebase.ts

```diff
@@ -141,76 +152,57 @@ export function createFirebaseEndpoints(emulator: StorageEmulator): Router {
   });
 
   const handleMetadataUpdate = async (req: Request, res: Response) => {
-    const md = storageLayer.getMetadata(req.params.bucketId, req.params.objectId);
-
-    if (!md) {
-      res.sendStatus(404);
-      return;
-    }
-
-    const decodedObjectId = decodeURIComponent(req.params.objectId);
-    const operationPath = ["b", req.params.bucketId, "o", decodedObjectId].join("/");
-
-    if (
-      !(await isPermitted({
-        ruleset: emulator.rules,
-        method: RulesetOperationMethod.UPDATE,
-        path: operationPath,
+    let metadata: StoredFileMetadata;
+    try {
+      metadata = await storageLayer.handleUpdateObjectMetadata({
+        bucketId: req.params.bucketId,
+        decodedObjectId: decodeURIComponent(req.params.objectId),
+        metadata: req.body as IncomingMetadata,
```

Comment:
We should avoid type assertions

https://g3doc.corp.google.com/javascript/typescript/g3doc/dev/readability/styleguide.md?cl=head#type-and-non-nullability-assertions